metricbeat/module/kubernetes/state_container/data.go

```diff
@@ -0,0 +1,88 @@
+package state_container
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/kubernetes/util"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+const (
+	// Nanocores conversion 10^9
+	nanocores = 1000000000
+)
+
+func eventMapping(families []*dto.MetricFamily) ([]common.MapStr, error) {
+	eventsMap := map[string]common.MapStr{}
+	for _, family := range families {
```

Comment:
I think it's pretty custom for this case, so we need to keep it.

I want to see what future metricsets using the helper have in common, perhaps we can still extract some helpers there